### PR TITLE
Remove reader API from Mqstream and replace it with consumer API

### PR DESCRIPTION
**Pull request by jaime0815**
issue: #15970 
Signed-off-by: yun.zhang <yun.zhang@zilliz.com>

### CODE REVIEW

**Comment by jaime0815** [comment]
depend with https://github.com/milvus-io/pulsar-client-go/pull/1  check-in

File: internal/mq/mqimpl/rocksmq/server/rocksmq_impl.go [hunk 1]
```diff
@@ -458,6 +458,39 @@ func (rmq *rocksmq) RegisterConsumer(consumer *Consumer) error {
...
+	if !ok {
+		log.Warn("RocksMQ: channel " + key + " not exists")
+		return ConsumerMessage{}, fmt.Errorf("ConsumerGroup %s, channel %s not exists", groupName, topicName)
+	}
+
+	readOpts := gorocksdb.NewDefaultReadOptions()
+	defer readOpts.Destroy()
+	iter := rocksdbkv.NewRocksIterator(rmq.store, readOpts)
+	defer iter.Close()
+	iter.SeekToLast()
```
Comment (xiaofan-luan) [inline internal/mq/mqimpl/rocksmq/server/rocksmq_impl.go:479]: The implementation here only works for single topic.
We need to add test cases for the following cases:
1) if a stream didn't write without any message, what is  the latest message return?
2)test with multiple topics.
Also the API should be GetLatestMsg(topicName) and it returned the latest written message for single topic.

File: internal/querynode/segment_loader.go [hunk 2]
```diff
@@ -517,48 +517,72 @@ func (loader *segmentLoader) FromDmlCPLoadDelete(ctx context.Context, collection
 	defer stream.Close()
 	pChannelName := rootcoord.ToPhysicalChannel(position.ChannelName)
 	position.ChannelName = pChannelName
-	stream.AsReader([]string{pChannelName}, fmt.Sprintf("querynode-%d-%d", Params.QueryNodeCfg.QueryNodeID, collectionID))
+	hasMore := stream.AsConsumerToLastMsg(pChannelName, fmt.Sprintf("querynode-%d-%d", Params.QueryNodeCfg.QueryNodeID, collectionID))
```
Comment (xiaofan-luan) [inline internal/querynode/segment_loader.go:520]: My proposal is to simply the interface with only one interface -> GetLatestMsg

	latestPosition = stream.GetLatestMsg 
	do AsConsumer
        do seek
	for _, tsMsg := range msgPack.Msgs {
		if tsMsg.Type() == commonpb.MsgType_Delete {
			dmsg := tsMsg.(*msgstream.DeleteMsg)
			if dmsg.CollectionID != collectionID {
				continue
			}
			log.Debug("delete pk",
				zap.Any("pk", dmsg.PrimaryKeys),
				zap.String("vChannelName", position.GetChannelName()),
				zap.Any("msg id", position.GetMsgID()),
			)
			processDeleteMessages(loader.historicalReplica, dmsg, delData)
		}

		if (tsMsg.Position() >= latestPosition) {
			hasMore = true
			break
		}
	}

File: internal/mq/msgstream/mq_msgstream.go [hunk 3]
```diff
@@ -167,38 +188,40 @@ func (ms *mqMsgStream) AsConsumerWithPosition(channels []string, subName string,
...
-				SubscriptionRolePrefix: subName,
-			})
-			if err != nil {
-				return err
-			}
-			if r == nil {
-				return errors.New("reader is nil")
-			}
+func (ms *mqMsgStream) hasMoreMsg(lastMsg MessageID) bool {
+	if lastMsg.PartitionIdx() <= 0 &&
```
Comment (xiaofan-luan) [inline internal/mq/msgstream/mq_msgstream.go:192]: the compare function should be implemented in MessageID

File: internal/mq/msgstream/msgstream.go [hunk 4]
```diff
@@ -55,22 +55,23 @@ type RepackFunc func(msgs []TsMsg, hashKeys [][]int32) (map[int32]*MsgPack, erro
...
-	HasNext(channelName string) bool
+
+	AsConsumer(channels []string, subName string)
+	AsConsumerWithPosition(channels []string, subName string, position mqwrapper.SubscriptionInitialPosition)
+	Chan() <-chan *MsgPack
 	Seek(offset []*MsgPosition) error
-	SeekReaders(msgPositions []*internalpb.MsgPosition) error
+
+	AsConsumerToLastMsg(channel string, subName string) bool
+	ReachLastMsg(position *internalpb.MsgPosition) (bool, error)
```
Comment (xiaofan-luan) [inline internal/mq/msgstream/msgstream.go:74]: We can replace AsConsumerToLastMsg  & ReachLastMsg with a simple interface named msgstream

**Pull request merged**